hashring.go

```diff
@@ -120,16 +120,12 @@ func (hr *HashRing) Locate(key string) (node string, err error) {
 
 	hkey, err := getHash(hr.hash, []byte(key))
 	if err != nil {
-		return node, fmt.Errorf("failed to fetch node: %v\n", err)
+		return node, fmt.Errorf("failed to fetch node: %v", err)
 	}
 
-	for _, k := range hr.idx {
-		if k < hkey {
-			continue
-		}
-
-		return hr.nodes[k], nil
+	pos := sort.Search(len(hr.idx), func(i int) bool { return hr.idx[i] < hkey })
```

Comment:
@pdaures Thank you for the PR. Since we are looking for key greater than equal hkey, `hr.idx[i] >= hkey` would give the correct node index.

https://play.golang.org/p/BedAJoKEjs